### DISCUSSION TITLE
Row level security is not applied to delete statements

### DISCUSSION THREAD

**Discussion by brianhv** [supabase/supabase#9581] [Questions] [answered]
There are a couple of statements under the "Notes" section on [this page](https://supabase.com/docs/reference/javascript/subscribe) that confuse me.

The first is the one in the subject, "Row level security is not applied to delete statements." Surely that doesn't mean that subscribing in javascript prevents RLS from checking DELETE statements, right? But I'm not sure what it does mean.

The second is "When a delete occurs, the contents of old_record will be broadcast to all subscribers to that table so ensure that each table's replica identity only contains information that is safe to expose publicly." Does that mean that the entire deleted row will be sent even to clients that shouldn't have access to that data? Or is it just saying the replica identity will be sent? And does the prior statement mean that the replica identity will be sent to everyone regardless of the RLS policies?

**Comment by GaryAustin1** [comment, marked as answer]
RLS will not apply to deletes.  Realtime has to read the table to see if RLS applies, but the row has been deleted so it can't.

You will get the replica identity sent to everyone.
From the github readme: 
`[OPTIONAL] If you want to receive the old record (previous values) on UPDATE and DELETE, you can set the REPLICA IDENTITY to FULL like this: ALTER TABLE your_table REPLICA IDENTITY FULL;. This has to be set for each table unfortunately.`

**Reply by brianhv** [reply]
Ah, so if I understand correctly:
* Postgres still applies RLS
* The realtime server, since the row is gone, has no way to evaluate RLS for the row. It therefore can't apply RLS itself, so falls back to sending deletes to all potentially relevant subscribers.
* These notifications contain only the replica identity to avoid leaking information about the deleted row. (And possibly because it doesn't get anything about the row besides its identity.)

**Reply by GaryAustin1** [reply]
Yes
